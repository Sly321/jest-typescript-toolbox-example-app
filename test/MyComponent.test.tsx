import * as React from "react"
import { shallow, ShallowWrapper } from "enzyme"

describe(`<MyComponent />`, () => {
  describe(`<div>asdf</div>`, () => {
    let tree: ShallowWrapper;

    beforeAll(() => {
      tree = shallow(<div className="reality">reality</div>);
    });

    it(`success test with enzyme matcher className`, () => {
      expect(tree.hasClass("reality")).toEqual(true);
    });

    it(`fail test with enzyme matcher text`, () => {
      expect(tree.text()).toEqual("expectation");
    });

    it(`fail test with enzyme matcher props`, () => {
      expect(tree.props()).toHaveProperty("expectation", "expectation");
    });

    it(`create a snapshot`, () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
