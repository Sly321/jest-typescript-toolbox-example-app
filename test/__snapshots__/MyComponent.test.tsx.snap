// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyComponent /> <div>asdf</div> create a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div
    className="reality"
  >
    reality
  </div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "reality",
      "className": "reality",
    },
    "ref": null,
    "rendered": "reality",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "reality",
        "className": "reality",
      },
      "ref": null,
      "rendered": "reality",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
